framework/app.mk

```diff
@@ -429,7 +429,18 @@ ifneq ($(app_test_LIB),)
 	lib_install_targets += $(dir $(app_test_LIB))install_lib_$(notdir $(app_test_LIB))
 endif
 
-install_libs: $(lib_install_targets)
+install_libs:: $(lib_install_targets)
```

Comment:
They are needed to avoid a rule replacement warning from make. Double colon rules. In this case the replace rule is always identical (but I still got a warning on my local machine), in the case below `install_data::` the `::` are needed to augment the existing rules step by step, adding new dependencies.

Reply:
Actually they are most likely not "always identical" but get overwritten with versions that have progressively more dependencies. The `::` approach avoids building the dependency list manually. I have no `data_install_targets`.